frontend/src/utils/userContext.js

```diff
@@ -27,24 +28,18 @@ const UserContextProvider = ({ children }) => {
         .then((user) => {
           setUser(user.data);
           setLoading(false);
+
+          guestRoutes.includes(location.pathname) && navigate("home");
         })
-        .catch((error) => {
-          console.log("An error occured");
-          if (
-            location.pathname === "/home" ||
-            location.pathname === "/profile" ||
-            location.pathname === "/profile/follows" ||
-            location.pathname === "/profile/edit"
-          ) {
-            navigate("login");
-          }
+        .catch(() => {
+          authenticatedRoutes.includes(location.pathname) && navigate("login");
         })
         .finally(() => {
           setLoading(false);
         });
     };
     fetchUser();
-  }, []);
+  }, [location.pathname, navigate]);
```

Comment:
Dont add any variable to watch here because it will trigger the fetchUser again which is not good when considering optimization

what you can do is to create another useEffect to check path and if has user or not since you already fetch the user